File: csharp/0271-encode-and-decode-strings.cs
public partial class Solution {
    public string Encode(List<string> strs) {
        var output = "";

        foreach (var word in strs) {
            output += $"{word.Length}#{word}";
        }

        return output;
    }

    public List<string> Decode(string s) {
        var output = new List<string>();
        var i = 0;

        while (i < s.Length) {
            var j = i;
            while (s[j] != '#') {
                j++;
            }
            var length = Int32.Parse(s.Substring(i, j - i));
            i = j + 1;
            j = i + length;
            output.Add(s.Substring(i, length));
            i = j;
        }

        return output;
    }
}

File: go/0271_encode_and_decode_strings.go
package main

import (
	"fmt"
	"strconv"
)

// time: 0(n)
// space: O(n)

func encode(input []string) string {
	output := ""

	for _, word := range input {
		output += fmt.Sprintf("%d#%s", len(word), word)
	}

	return output
}

func decode(input string) []string {
	var output []string
	i := 0

	for i < len(input) {
		j := i
		for input[j] != '#' {
			j++
		}
		length, _ := strconv.Atoi(input[i:j])
		i = j + 1
		j = i + length
		output = append(output, input[i:j])
		i = j
	}

	return output
}
